pkg/image/controller/controller.go

```diff
@@ -39,35 +33,105 @@ const retryCount = 2
 // the image stream is not modified (so it will be tried again later). If a permanent
 // failure occurs the image is marked with an annotation. The tags of the original spec image
 // are left as is (those are updated through status).
+// There are 3 use cases here:
+// 1. spec.DockerImageRepository defined without any tags results in all tags being imported
+//    from default registry
+// 2. spec.DockerImageRepository + tags defined - import only the provided tags,
+//    for kind == DockerImage import underlying image, iow. exact tag (no provided means latest),
+//    for kind != DockerImage import tag from the default registry
+// 3. spec.DockerImageRepository not defined - import tags per its definition.
 func (c *ImportController) Next(stream *api.ImageStream) error {
 	if !needsImport(stream) {
 		return nil
 	}
+
 	name := stream.Spec.DockerImageRepository
+	insecure := stream.Annotations != nil && stream.Annotations[api.InsecureRepositoryAnnotation] == "true"
```

Comment:
You don't need to check for nil - retrieving an item from a nil map returns `<default value for map value type>, false`